L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,256 @@
+  std::vector<uint32_t> tc_coord_v = TCcoord_uv.getParameter<std::vector<uint32_t>>("TCv");
+  if (tc_coord_u.size() != tc_coord_v.size())
+    throw cms::Exception("BadParameter") << "TCu and TCv vectors should be of same size";
+  for (size_t i = 0; i < tc_coord_u.size(); ++i)
+    tc_coord_uv_.emplace(std::make_pair(tc_coord_u.at(i), tc_coord_v.at(i)), i);
+
+  // Get truncation parameters
+  const edm::ParameterSet& truncationParamConfig = conf.getParameterSet("BackendStage1Params");
+  roz_min_ = truncationParamConfig.getParameter<double>("rozMin");
+  roz_max_ = truncationParamConfig.getParameter<double>("rozMax");
+  roz_bins_ = truncationParamConfig.getParameter<uint32_t>("rozBins");
+  max_tcs_per_bins_ = truncationParamConfig.getParameter<std::vector<uint32_t>>("maxTcsPerBin");
+  phi_edges_ = truncationParamConfig.getParameter<std::vector<double>>("phiSectorEdges");
+}
+
+HGCalBackendStage1ParameterExtractor::~HGCalBackendStage1ParameterExtractor() {}
+
+void HGCalBackendStage1ParameterExtractor::beginRun(const edm::Run& /*run*/, const edm::EventSetup& es) {
+  triggerGeometry_ = es.getHandle(triggerGeomToken_);
+
+  json outJSON;
+
+  // fill MetaData
+  outJSON["MetaData"]["GeometryVersion"] = detector_version_;
+  outJSON["MetaData"]["fpgaId"] = the_fpga_;
+
+  // fill geometry configuration
+  outJSON["TriggerGeometryConfig"]["TriggerGeometryName"] = trigger_geom_;
+  outJSON["TriggerGeometryConfig"]["DisconnectedLayers"] = disconnected_layers_;
+  outJSON["TriggerGeometryConfig"]["JsonMappingFile"] = json_mapping_file_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorTriggerCellSize"] = scintillator_trigger_cell_size_;
+
+  // fill truncation parameters
+  outJSON["TruncationConfig"]["rozMin"] = roz_min_;
+  outJSON["TruncationConfig"]["rozMax"] = roz_max_;
+  outJSON["TruncationConfig"]["rozBins"] = roz_bins_;
+  outJSON["TruncationConfig"]["maxTcsPerBin"] = max_tcs_per_bins_;
+  outJSON["TruncationConfig"]["phiSectorEdges"] = phi_edges_;
+
+  // fill trigger geometry
+  fillTriggerGeometry(outJSON);
+
+  // write out JSON file
+  std::ofstream outputfile(outJSONname_.c_str());
+  outputfile << std::setw(4) << outJSON << std::endl;
+}
+
+void HGCalBackendStage1ParameterExtractor::fillTriggerGeometry(json& json_file) {
+  std::vector<uint32_t> trigger_cells;
+
+  // retrieve valid trigger cells
+  for (const auto& id : triggerGeometry_->eeGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (triggerGeometry_->eeTopology().valid(id))
+      trigger_cells.push_back(triggerGeometry_->getTriggerCellFromCell(id));
+  }
+  for (const auto& id : triggerGeometry_->hsiGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (triggerGeometry_->hsiTopology().valid(id))
+      trigger_cells.push_back(triggerGeometry_->getTriggerCellFromCell(id));
+  }
+  for (const auto& id : triggerGeometry_->hscGeometry()->getValidDetIds()) {
+    trigger_cells.push_back(triggerGeometry_->getTriggerCellFromCell(id));
+  }
```

Comment:
There will be duplications of TCs since there are several cells in one TC.
To avoid duplication automatically, a `std::unordered_set` could be used instead of a `std::vector`